src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -253,12 +275,20 @@ def process_delayed_workflows(
     workflow_event_dcg_data = fetch_group_to_event_data(project_id, Workflow, batch_key)
 
     # Get mappings from DataConditionGroups to other info
-    dcg_to_groups, dcg_to_workflow, _ = get_dcg_group_workflow_data(workflow_event_dcg_data)
-    workflow_ids_to_workflows, workflows_to_envs = fetch_workflows_and_environments(
-        list(dcg_to_workflow.values())
+    dcg_to_groups, dcg_to_workflow, dcg_to_detector, _ = get_dcg_group_workflow_detector_data(
+        workflow_event_dcg_data
+    )
+    workflow_ids_to_workflows, detector_ids_to_detectors, workflows_to_envs = (
+        fetch_workflows_detectors_envs(
+            list(dcg_to_workflow.values()), list(dcg_to_detector.values())
+        )
     )
     data_condition_groups = fetch_active_data_condition_groups(
```

Comment:
🤔 do we need to filter this far on the data condition groups? can we just use the list of ids from the buffer? (since those are already filtered / validated etc)

Reply:
are we assuming the detector/workflow is already enabled and it won't change by the time we are processing this? if it's delayed there could be a mismatch between when the user disables the detector/workflow and when we fire the alert

Reply:
chatted about this, i'm removing it because we're making extra db queries for an uncommon edge case